discovery.cpp

```diff
@@ -359,13 +359,21 @@ void DeRestPluginPrivate::internetDiscoveryExtractVersionInfo(QNetworkReply *rep
 
             if (!version.isEmpty())
             {
-                if (gwUpdateVersion != version)
+                QStringList gwUpdateVersionList = gwUpdateVersion.split('.');
+                QStringList versionList = version.split('.');
+                if (((gwUpdateVersionList[0].toInt() <  versionList[0].toInt())) ||

```

Comment:
I'd recommend to add a check that both lists contain at least 3 values:

    (gwUpdateVersionList.size() >= 3 && versionList.size() >= 3)

Reply:
Thanks, done